server/task/src/main/java/org/apache/james/task/MemoryWorkQueue.java

```diff
@@ -20,22 +20,35 @@
 package org.apache.james.task;
 
 import java.io.IOException;
+import java.util.Optional;
 import java.util.concurrent.LinkedBlockingQueue;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 import reactor.core.Disposable;
 import reactor.core.publisher.Mono;
 import reactor.core.scheduler.Schedulers;
 
 public class MemoryWorkQueue implements WorkQueue {
 
+    public static final Logger LOGGER = LoggerFactory.getLogger(MemoryWorkQueue.class);
     private final TaskManagerWorker worker;
     private final Disposable subscription;
     private final LinkedBlockingQueue<TaskWithId> tasks;
 
     public MemoryWorkQueue(TaskManagerWorker worker) {
         this.worker = worker;
         this.tasks = new LinkedBlockingQueue<>();
```

Comment:
The blocking queue should not be used at all, we should rather use a processor (see https://projectreactor.io/docs/core/release/reference/index.html#_safely_produce_from_multiple_threads_by_using_the_sink_facade , we already use WorkQueueProcessor in another part of in-memory code)